### PR TITLE
enhance template fuzz testing

### LINKED ISSUE #5776: Add fuzz tests for templates
This is the current fuzzer for templates:

```
        func(j *template.Template, c fuzz.Continue) {
            c.Fuzz(&j.ObjectMeta)
            c.Fuzz(&j.Parameters)
            // TODO: replace with structured type definition
            j.Objects = []runtime.Object{}
        },
```

We need to test conversions with templates, especially since they make use of unstructured API objects, which is uncommon.

### PR DESCRIPTION
**Pull request by jim-minter**
fixes #5776

### CODE REVIEW

**Comment by jim-minter** [comment]
@liggitt I've found it a bit of a case of 1000 monkeys and 1000 typewriters writing this, so I'd appreciate your review.  I think this would have caught #5557 - does it match what you were looking for?

**Comment by jim-minter** [comment]
@liggitt bump

**Comment by bparees** [comment]
@liggitt re-bump

File: pkg/api/serialization_test.go [hunk 1]
```diff
@@ -169,10 +174,43 @@ func originFuzzer(t *testing.T, seed int64) *fuzz.Fuzzer {
...
+				}
+				b = bytes.TrimRight(b, "\n")
+
+				j.Objects = append(j.Objects,
+					&runtime.Unknown{
+						ContentType: "application/json",
+						Raw:         b,
+					},
+				)
+			}
```
Comment (liggitt) [inline pkg/api/serialization_test.go]: could also add an unknown type to make sure we can round trip it. something like:

```go
j.Objects = append(j.Objects, &runtime.Unknown{
	ContentType: "application/json",
	Raw:         []byte(`{"kind":"Foo","apiVersion":"mygroup/v1","complex":{"a":true},"list":["item"],"bool":true,"int":1,"string":"hello"}`),
})
```

**Comment by jim-minter** [comment]
unknown type added, thanks, please merge

**Pull request merged**